src/effects/effectslot.cpp

```diff
@@ -150,72 +312,85 @@ EffectButtonParameterSlotPointer EffectSlot::getEffectButtonParameterSlot(unsign
     return m_buttonParameters[slotNumber];
 }
 
-void EffectSlot::loadEffect(EffectPointer pEffect, bool adoptMetaknobPosition) {
-    //qDebug() << debugString() << "loadEffect"
-    //         << (pEffect ? pEffect->getManifest().name() : "(null)");
-    if (pEffect) {
-        m_pEffect = pEffect;
-        m_pControlLoaded->forceSet(1.0);
-        m_pControlNumParameters->forceSet(pEffect->numKnobParameters());
-        m_pControlNumButtonParameters->forceSet(pEffect->numButtonParameters());
+bool EffectSlot::loadEffect(EffectManifestPointer pManifest, EffectInstantiatorPointer pInstantiator,
+        const QSet<ChannelHandleAndGroup>& activeChannels) {
+    // Remove an effect if already loaded in slot
+    if (m_pEngineEffect != nullptr) {
+        unloadEffect();
+    }
 
-        // The enabled status persists in the EffectSlot when loading a new
-        // EffectPointer to the EffectSlot. Effects and EngineEffects default to
-        // disabled, so if this EffectSlot was enabled, enable the Effect and EngineEffect.
-        pEffect->setEnabled(m_pControlEnabled->toBool());
+    VERIFY_OR_DEBUG_ASSERT(pManifest != nullptr && pInstantiator != nullptr) {
```

Comment:
This prevents effects from being unloaded. This shouldn't be a debug assertion. In this case, call `unloadEffect()` then return. Since `unloadEffect()` is required when replacing an effect as well, you can start the function with just `unloadEffect()` outside of any `if` block, then return if the manifest is a nullptr.